arrow-libs/optics/arrow-optics/src/commonMain/kotlin/arrow/optics/Fold.kt

```diff
@@ -327,3 +305,18 @@ public interface Fold<S, A> {
       Every.tuple10()
   }
 }
+
+/**
+ * Fold using the given [empty] element and [combine].
+ */
+public fun <S, A> Fold<S, A>.fold(empty: A, combine: (A, A) -> A, source: S): A =
+  foldMap(empty, combine, source, ::identity)
+
+/**
+ * Map each target to a type [R] and combine the results as a fold.
+ */
+public fun <S, A, R> Fold<S, A>.foldMap(empty: R, combine: (R, R) -> R, source: S, map: (focus: A) -> R): R =
```

Comment:
Is this an extension because it was a conflicting API? and `@JvmName` doesn't work on `interface`?

Reply:
As discussed above, this is an extension so that the method is not added to the interface, which would break binary compatibility.

Reply:
Adding methods is not a problem, only removal, since this would break compatibility for users relying on dependencies with older Arrow versions of the 1.x.x series.

Reply:
Oh, I misunderstood the binary compatibility then. I've moved the functions back to `Fold`, where they should logically belong.

Reply:
It's really a JVM thing I guess, the problem is the following:

 - Library A has a dependency on Arrow 1.1.5 relies on `Fold.foldMap`
 - Project B has dependency on Library A & has dependency on Arrow 1.1.5
 - Arrow 1.2.0 removes `Fold.foldMap`
 - Project B bumps dependency to Arrow 1.2.0 => Fold.foldMap: [MethodNotFoundException](https://docs.oracle.com/javaee/7/api/javax/faces/el/MethodNotFoundException.html) at runtime.

So adding methods is non-problematic.